astropy/io/misc/yaml.py

```diff
@@ -138,6 +141,11 @@ def _ndarray_constructor(loader, node):
     # construct_sequence.
     map = loader.construct_mapping(node, deep=True)
     map["buffer"] = base64.b64decode(map["buffer"])
+
+    if "object" in map["dtype"]:
```

Comment:
Why is this written `"object" in map["dtype"]` instead of `map["dtype"] == "object"`?

Reply:
I was being overly cautious 😅 , accounting for possible dtypes that have "object" in the string. If it's not likely, it's surely better to change it to the exact match.

Reply:
There is only `object`: https://numpy.org/doc/stable/reference/arrays.scalars.html. If we were being supremely cautious then we would make an allowlist of all accepted dtypes instead of this (single-element) denylist. But I think here it's OK to just do the single exact match to exclude `object`.

Reply:
Done